app/serializers/api/admin/index_order_cycle_serializer.rb

```diff
@@ -61,6 +61,10 @@ def delete_path
         admin_order_cycle_path(object)
       end
 
+      def subscriptions_count
+        ProxyOrder.not_canceled.where(order_cycle_id: object.id).count
```

Comment:
have you checked this doesn't add an N+1?

Reply:
This `count` is on database level. As long as we use indexes for cancelled and order cycle, this should be really fast. I just checked and we don't have an index for `cancelled_at`. For this purpose, we don't need to index all date values. We just want to know if it's `null` or set. I guess we can limit the index with `length: 4` which includes the year.

Reply:
If someone had 100,000 standing orders with 12 weeks of proxy orders ahead for one order cycle, the records exceed a million and this would become really important.

Reply:
Maybe the issue is more that this query will generate a new query for each order cycle in the collection when the index page is rendered?

Maybe if I defined these associations on `OrderCycle`:
````ruby
has_many :active_proxy_orders, class_name: 'ProxyOrder', conditions: { where('proxy_orders.canceled_at IS NULL') }
has_many :subscriptions, through: :active_proxy_orders
````
Then I could preload the subscriptions associated with all of the order cycles in the collection, and reduce the number of queries required to count subscriptions from N to 1?

That would make this method become:
````ruby
def subscriptions_count
    object.subscriptions.count
end
````
Which would hopefully be an array count, rather than a db count.

Thoughts?

Reply:
I was thinking that db counts are much quicker than array counts. Loading all the subscriptions into objects just to count them sounds very inefficient. If you want to load the count with the query, you probably have to create a scope in which you modify the `select` with your own join and group_by. I'm not sure if there is an ActiveRecords technique for that.

Reply:
> I was thinking that db counts are much quicker than array counts.

Yes definitely when only one order cycle is being serialized, but I think if you serializing 30 order cycles then preloading the subscriptions and counting the ones that apply to each order cycle in ruby is going to be faster than running 30 separate queries (one for each OC)? It's probably not a problem at this stage. I think we should just stick with it the way it is.

Reply:
I don't know the implementation of ActiveRecord well enough. It could be that ActiveRecord loads everything from one query, but still creates independent objects. I don't know.

We can either investigate this and test what's actually faster, or we can leave it as is and find out when we have a performance issue.

@sauloperez What's your view? Should we invest more work to find out or can this go through as is?

Reply:
It's hard for me to accept the fact that we introduce a clear N+1 query buuuut I agree with 

> we can leave it as is and find out when we have a performance issue.

We have way too many things on our plate, so let's be pragmatic and fix it when it becomes a bottleneck.

Reply:
@sauloperez where in the N+1. This `count` is uses a db `COUNT`, we're not loading the objects into ruby. Am I missing something? I'm really happy to fix it if I know what to do...

Reply:
So, if I guessed correctly by the name of this serializer (`IndexOrderCycleSerializer`), we use it to render a list of order cycles. To do that, this serializer gets instantiated for each order cycle we fetched from DB, issuing a query each time `#subscriptions_count` is called.

There we have it; 1 query to fetch N order cycles and N queries to fetch the count of subscriptions. Am I right or did I totally misunderstood how this serializer works?

Reply:
Note for posterity: we decided to resolve this by doing the counts in a single query in the controller and then pass them to the serializer.